services/services.go

```diff
@@ -31,6 +32,11 @@ import (
 	"github.com/m3db/m3cluster/shard"
 )
 
+const (
+	defaultLeaderTimeout = 30 * time.Second
```

Comment:
hm that's a really long timeout, perhaps change it to something like 10 seconds? Same goes for the resign timeout.